test/test_join.js

```diff
@@ -0,0 +1,21 @@
+/* eslint-env mocha */
+'use strict'
+
+const { assert } = require('chai')
+const shlex = require('../shlex')
+
+describe('shlex.join()', () => {
+  const posixTestcases = [
+    [["ls", "-al", '/'], 'ls -al /'],
+    [[ 'rm', '-f', '/Volumes/Macintosh HD' ], "rm -f '/Volumes/Macintosh HD'"],
+  ]
+
+  it('should join according to POSIX rules', () => {
+    posixTestcases.forEach(function (test) {
+      const input = test[0]
+      const expected = test(1)
```

Comment:
Any reason for parentheses like `test(1)` instead of `test[1]`? Or use destructuring like 

```javascript
const [input, expected] = test;
```